File: pset6/readability/readability.py
from cs50 import get_string


def main():
    s = get_string("Text: ")
    
    # Initializes variables
    # Words initialized at 1 to account for the last word
    countLetters = 0
    countWords = 1
    countSentences = 0
    
    # Iterate over each character in text
    for c in s:
        if c is " ":
            countWords += 1
        elif c.isalpha():
            countLetters += 1
        elif c == "." or c == "!" or c == "?":
            countSentences += 1
    
    L = (float(countLetters) / countWords) * 100
    S = (float(countSentences) / countWords) * 100
    
    # Calculate index
    index = round(0.0588 * L - 0.296 * S - 15.8)
    
    # Find grade
    if index < 1:
        print("Before Grade 1")
        return 0
    if index < 16:
        print(f"Grade {index}")
        return 0
    if index >= 16:
        print("Grade 16+")
        return 0


main()

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string s);
int count_words(string s);
int count_sentences(string s);


// Readability function
int main(void) 
{
    // Input text from user
    string text = get_string("Text: ");
    
    // Number of letters, words and sentences
    int L = count_letters(text);
    int W = count_words(text);
    int S = count_sentences(text);
    
    printf("%i letters %i words %i sentences", L W S);
    // Index computation  
    int grade = round(0.0588 * ((float) L / W) * 100 - 0.296 * ((float) S / W) * 100 - 15.8); 

    // Print grade
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    } 
    else if (grade >= 16) 
    {
        printf("Grade 16+\n");
    } 
    else 
    {
        printf("Grade %d\n", grade);
    }
}



// Count letters function
int count_letters(string s)
{
    int L = 0;
    for (int i = 0, n = strlen(s); i < n; i++) 
    {
        if ((s[i] >= 'a' && s[i] <= 'z')
            || (s[i] >= 'A' && s[i] <= 'Z'))
        {
            L++;
        }
    }
    return L;
}

// Count words function
int count_words(string s)
{
    int W = 1;
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if (s[i] == ' ')
        {
            W++;
        }
    }
    return W;
}

// Count sentences function
int count_sentences(string s)
{
    int S = 0;
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if (s[i] == '.' || s[i] == '!' || s[i] == '?')
        {
            S++;
        }
    }
    return S;
}